src/index.ts

```diff
@@ -40,22 +41,23 @@ export class AzureDataStore {
   public constructor (path: string, opts: AzureDSInputOptions) {
     this.path = path;
     this.container = opts.containerName;
-    this.blobService = !opts.connectionString ? storage.createBlobService() : storage.createBlobService(opts.connectionString);
+    if (!AzureDataStore.blobService) {
+      AzureDataStore.blobService = !opts.connectionString ? storage.createBlobService() : storage.createBlobService(opts.connectionString);
```

Comment:
What does it mean for `createBlobService()` without a connection string? Seems like throwing an error would make sense in that case?

Reply:
we can either pass the connection string in the method, otherwise it will try to use the environment variable 'AZURE_STORAGE_CONNECTION_STRING'